$OpenBSD$

Make setSocketKeepAlive a no-op on OpenBSD. Kernel does not support
TCP_KEEPIDLE/TCP_KEEPINTVL/TCP_KEEPCNT socket options (ENOPROTOOPT).

--- src/Simplex/Messaging/Transport/KeepAlive.hs.orig
+++ src/Simplex/Messaging/Transport/KeepAlive.hs
@@ -64,8 +64,13 @@

-setSocketKeepAlive :: Socket -> KeepAliveOpts -> IO ()
-setSocketKeepAlive sock KeepAliveOpts {keepCnt, keepIdle, keepIntvl} = do
-  setSocketOption sock KeepAlive 1
-  setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPIDLE) keepIdle
-  setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPINTVL) keepIntvl
-  setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPCNT) keepCnt
+#if defined(openbsd_HOST_OS)
+setSocketKeepAlive :: Socket -> KeepAliveOpts -> IO ()
+setSocketKeepAlive _ _ = pure ()
+#else
+setSocketKeepAlive :: Socket -> KeepAliveOpts -> IO ()
+setSocketKeepAlive sock KeepAliveOpts {keepCnt, keepIdle, keepIntvl} = do
+  setSocketOption sock KeepAlive 1
+  setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPIDLE) keepIdle
+  setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPINTVL) keepIntvl
+  setSocketOption sock (SockOpt _SOL_TCP _TCP_KEEPCNT) keepCnt
+#endif
